### PR TITLE
update dray package with new versions and to fix issue with test variant

### PR DESCRIPTION
**Pull request by cyrush**

### CODE REVIEW

File: var/spack/repos/builtin/packages/dray/package.py [hunk 1]
```diff
@@ -33,7 +33,9 @@ class Dray(Package, CudaPackage):
 
     maintainers = ['mclarsen', 'cyrush']
 
-    version('develop', branch='develop', submodules='True')
+    version('develop',  branch='develop', submodules='True', preferred=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/dray/package.py:36]: I'm hesitant to make a non-checksummable moving target the default installation version...

Comment (cyrush) [reply var/spack/repos/builtin/packages/dray/package.py:36]: if something goes wrong, you can point any angry folks at me :-) ?

**Comment by cyrush** [comment]
@adamjstewart  would it be possible to get this merged? The fact that this was outstanding and it atrophied our process update to newer spack.

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/dray/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/dray/package.py:36]: @scheibelp I merged this for now, but add "defaulting to non-checksummable versions" to your security policy discussion. We should have an official policy on this.

**Pull request merged**